File: app/src/main/java/com/nfcalarmclock/audio/options/NacAlarmAudioOptionsDialog.java
package com.nfcalarmclock.audio.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import androidx.fragment.app.DialogFragment;

import com.nfcalarmclock.R;

/**
 */
public class NacAlarmAudioOptionsDialog
	extends DialogFragment
{

	/**
	 * Listener for when an audio option is clicked.
	 */
	@SuppressWarnings("UnnecessaryInterfaceModifier")
	public interface OnAudioOptionClickedListener
	{
		public void onAudioOptionClicked(long alarmId, int which);
	}

	/**
	 * Tag for the class.
	 */
	public static final String TAG = "NacAlarmAudioOptionsDialog";

	/**
	 * Alarm ID.
	 */
	private long mAlarmId;

	/**
	 * Listener for when an audio option is clicked.
	 */
	private OnAudioOptionClickedListener mOnAudioOptionClickedListener;

	/**
	 * Call the OnAudioOptionClickedListener object, if it has been set.
	 *
	 * @param  which  Which item was clicked.
	 */
	public void callOnAudioOptionClickedListener(int which)
	{
		OnAudioOptionClickedListener listener =
			this.getOnAudioOptionClickedListener();
		long alarmId = this.getAlarmId();

		if (listener != null)
		{
			listener.onAudioOptionClicked(alarmId, which);
		}
	}

	/**
	 * Get the alarm ID.
	 *
	 * @return The alarm ID.
	 */
	public long getAlarmId()
	{
		return this.mAlarmId;
	}

	/**
	 * Get the OnAudioOptionClickedListener object.
	 *
	 * @return The OnAudioOptionClickedListener object.
	 */
	public OnAudioOptionClickedListener getOnAudioOptionClickedListener()
	{
		return this.mOnAudioOptionClickedListener;
	}

	/**
	 */
	@NonNull
	@Override
	public Dialog onCreateDialog(@Nullable Bundle savedInstanceState)
	{
		return new AlertDialog.Builder(requireContext())
			.setTitle(getString(R.string.title_audio_option))
			.setItems(R.array.audio_options, (dialog, which) ->
				this.callOnAudioOptionClickedListener(which))
			.create();
	}

	/**
	 * Set the alarm ID.
	 *
	 * @param  id  Alarm ID.
	 */
	public void setAlarmId(long id)
	{
		this.mAlarmId = id;
	}

	/**
	 * Set the OnAudioOptionClickedListener object.
	 *
	 * @param  listener  The OnAudioOptionClickedListener object.
	 */
	public void setOnAudioOptionClickedListener(
		OnAudioOptionClickedListener listener)
	{
		this.mOnAudioOptionClickedListener = listener;
	}

}

File: app/src/main/java/com/nfcalarmclock/audiooptions/NacAlarmAudioOptionsDialog.kt
package com.nfcalarmclock.audiooptions

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import com.nfcalarmclock.R
import com.nfcalarmclock.view.dialog.NacDialogFragment

/**
 * Show the audio options of the alarm.
 */
class NacAlarmAudioOptionsDialog
	: NacDialogFragment()
{

	/**
	 * Listener for when an audio option is clicked.
	 */
	fun interface OnAudioOptionClickedListener
	{
		fun onAudioOptionClicked(alarmId: Long, which: Int)
	}

	/**
	 * Alarm ID.
	 */
	var alarmId: Long = 0

	/**
	 * Listener for when an audio option is clicked.
	 */
	var onAudioOptionClickedListener: OnAudioOptionClickedListener? = null

	/**
	 * Called when the dialog is created.
	 */
	override fun onCreateDialog(savedInstanceState: Bundle?): Dialog
	{
		// Setup the shared preferences
		setupSharedPreferences()

		// Create the dialog
		return AlertDialog.Builder(requireContext())
			.setTitle(R.string.title_audio_option)
			.setItems(R.array.audio_options) { _, which: Int ->

				// Call the listener
				onAudioOptionClickedListener?.onAudioOptionClicked(alarmId, which)

			}
			.create()
	}

	companion object
	{

		/**
		 * Tag for the class.
		 */
		const val TAG = "NacAlarmAudioOptionsDialog"

	}

}